app/presenters/region_presenter.rb

```diff
@@ -94,22 +94,28 @@ def sources_per_jurisdiction
   end
 
   def protected_areas_per_iucn_category
-    protected_areas_per_iucn_category_array = []
 
-    @countries.each do |country|
-      country.protected_areas_per_iucn_category.each do |protected_areas_per_iucn|
-        hash =  {
-          "#{protected_areas_per_iucn['iucn_category_id']}" => {
-            iucn_category_name: protected_areas_per_iucn["iucn_category_name"],
-            count: protected_areas_per_iucn["count"],
-            percentage: protected_areas_per_iucn["percentage"]
-          }
-        }
-        protected_areas_per_iucn_category_array << hash
+    valid_iucn_categories = IucnCategory.all.pluck(:name, :id).uniq
+    region_data = Hash.new { |hash, key| hash[key] = Hash.new }
+    processed_data = []
+
+    region.countries.each do |country|
+      country.protected_areas_per_iucn_category.each do |protected_area|
+        region_data["#{protected_area['iucn_category_name']}"][:count] = Array.new if region_data["#{protected_area['iucn_category_name']}"][:count].nil?
```

Comment:
It looks good!
What I would probably do to further improve the readability of this function, is to create a variable in the loop to store the value of `protected_area['iucn_category_name']`.
In this way you don't need to access the protected_area hash every time and it's easier to read with just a variable.

Also, I would replace `Array.new` with just `[]`

Reply:
Thanks @Levia I will update the code in the way you have suggested!